### PR TITLE
Upgrade golang to 1.13.6

### PR DESCRIPTION
**Pull request by sayboras**
Update the latest patch version for golang

### CODE REVIEW

File: hack/jenkins/common.sh [hunk 1]
```diff
@@ -31,7 +31,8 @@ export KUBECONFIG="${TEST_HOME}/kubeconfig"
 export PATH=$PATH:"/usr/local/bin/:/usr/local/go/bin/:$GOPATH/bin"
 
 # installing golang so we could do go get for gopogh
-sudo ./installers/check_install_golang.sh "1.13.4" "/usr/local" || true
+WANT_GOLANG_VERSION=$(grep '^GO_VERSION' Makefile | awk '{ print $3 }')
+sudo ./installers/check_install_golang.sh "$WANT_GOLANG_VERSION" "/usr/local" || true
```
Comment (medyagh) [inline hack/jenkins/common.sh]: there is another shell script with 
also change the version in this file:
hack/jenkins/minikube_cross_build_and_upload.sh

Comment (sayboras) [reply hack/jenkins/common.sh]: oh this script is retrieving golang version from Makefile directly.

```bash
WANT_GOLANG_VERSION=$(grep '^GO_VERSION' Makefile | awk '{ print $3 }')
./hack/jenkins/installers/check_install_golang.sh $WANT_GOLANG_VERSION /usr/local
```

File: hack/jenkins/common.sh [hunk 2]
```diff
@@ -31,7 +31,8 @@ export KUBECONFIG="${TEST_HOME}/kubeconfig"
 export PATH=$PATH:"/usr/local/bin/:/usr/local/go/bin/:$GOPATH/bin"
 
 # installing golang so we could do go get for gopogh
-sudo ./installers/check_install_golang.sh "1.13.4" "/usr/local" || true
+WANT_GOLANG_VERSION=$(grep '^GO_VERSION' Makefile | awk '{ print $3 }')
```
Comment (medyagh) [inline hack/jenkins/common.sh]: This line doesn't work in jenkins, in Jenkins job configurations we do not copy the Makefile to the agent that runs the e2e (only the e2e and the scripts)

Comment (sayboras) [reply hack/jenkins/common.sh]: Thanks :100:

**Comment by medyagh** [comment]
on jenkins I see this error :

```

17:12:32 + true
17:12:32 + make -j 16 all
17:12:32 docker run --rm -e GOCACHE=/app/.cache -e IN_DOCKER=1 --user 108:113 -w /app -v /var/lib/jenkins/go/src/k8s.io/minikube:/app -v /var/lib/jenkins/go:/go --init k8s.gcr.io/kube-cross:v1.13.8-1 /bin/bash -c '/usr/bin/make pkg/minikube/assets/assets.go'
17:12:32 docker run --rm -e GOCACHE=/app/.cache -e IN_DOCKER=1 --user 108:113 -w /app -v /var/lib/jenkins/go/src/k8s.io/minikube:/app -v /var/lib/jenkins/go:/go --init k8s.gcr.io/kube-cross:v1.13.8-1 /bin/bash -c '/usr/bin/make pkg/minikube/translate/translations.go'
17:12:32 Unable to find image 'k8s.gcr.io/kube-cross:v1.13.8-1' locally
17:12:32 Unable to find image 'k8s.gcr.io/kube-cross:v1.13.8-1' locally
17:12:32 docker run --rm -e GOCACHE=/app/.cache -e IN_DOCKER=1 \
17:12:32 	--user 108:113 -w /app \
17:12:32 	-v /var/lib/jenkins/go/src/k8s.io/minikube:/app -v /var/lib/jenkins/go:/go --init --entrypoint "" \
17:12:32 	karalabe/xgo-1.12.x /bin/bash -c 'CC=o64-clang CXX=o64-clang++ /usr/bin/make out/docker-machine-driver-hyperkit'
17:12:32 docker inspect -f '{{.Id}} {{.RepoTags}}' gcr.io/k8s-minikube/kvm-build-image:1.13.8 || make kvm-image
17:12:33 Pulling repository k8s.gcr.io/kube-cross
17:12:33 Pulling repository k8s.gcr.io/kube-cross
17:12:33 docker: Error: Status 405 trying to pull repository kube-cross: "v1 Registry API is disabled. If you are not explicitly using the v1 Registry API, it is possible your v2 image could not be found. Verify that your image is available, or retry with `dockerd --disable-legacy-registry`. See https://cloud.google.com/container-registry/docs/support/deprecation-notices".
17:12:33 See 'docker run --help'.
17:12:33 Makefile:280: recipe for target 'pkg/minikube/assets/assets.go' failed
17:12:33 make: *** [pkg/minikube/assets/assets.go] Error 125
17:12:33 make: *** Waiting for unfinished jobs....
17:12:33 docker: Error: Status 405 trying to pull repository kube-cross: "v1 Registry API is disabled. If you are not explicitly using the v1 Registry API, it is possible your v2 image could not be found. Verify that your image is available, or retry with `dockerd --disable-legacy-registry`. See https://cloud.google.com/container-registry/docs/support/deprecation-notices".
17:12:33 See 'docker run --help'.
17:12:33 Makefile:292: recipe for target 'pkg/miniku
```

it seems like we use k8s.gcr.io/kube-cross to build the minikube binaries.
and 
kvm-build-image:1.13.8 is not available yet !

**Comment by medyagh** [comment]
Basically means we have to wait for kubernetes project to to make golang image.

We can bump the golang to the latest kubecross image available

**Comment by sayboras** [comment]
> Basically means we have to wait for kubernetes project to to make golang image.
> 
> We can bump the golang to the latest kubecross image available

Thanks, I added `WIP` in PR to avoid merging

**Comment by afbjorklund** [comment]
You can upgrade to go1.13.6 if you want (that is the version `v1.18.0-alpha.5` is using)

File: Makefile [hunk 3]
```diff
@@ -29,7 +29,7 @@ DEB_VERSION ?= $(subst -,~,$(RAW_VERSION))
 RPM_VERSION ?= $(DEB_VERSION)
 
 # used by hack/jenkins/release_build_and_upload.sh and KVM_BUILD_IMAGE, see also BUILD_IMAGE below
-GO_VERSION ?= 1.13.4
+GO_VERSION ?= 1.13.8
```
Comment (afbjorklund) [inline Makefile]: 1.13.6

See https://github.com/kubernetes/kubernetes/tree/master/build/build-image/cross

**Comment by afbjorklund** [comment]
Go 1.13.8 is now available, as a milestone for Kubernetes 1.18

**Pull request merged**